src/Campaigns/ServiceProvider.php

```diff
@@ -229,11 +229,14 @@ private function registerCampaignBlocks()
     }
 
     /**
+     * @unreleased Move to admin_init hook
      * @since 4.0.0
      */
     private function loadCampaignOptions()
     {
-        Hooks::addAction('init', LoadCampaignOptions::class);
+        if (CampaignsAdminPage::isShowingDetailsPage()) {
+            Hooks::addAction('admin_init', LoadCampaignOptions::class);
+        }
```

Comment:
@pauloiankoski couple things here:

1. We should always be accessing php globals inside an action or else this will get checked on every page load since this is the boot of the service provider - so I recommend putting those conditionals _inside_ the action call to scope their usage.

```php
add_action('hook', function(){
    if ($_GET['foo'] === 'bar') {
        // do something
    }
});
```
2. If these are specific to loading scripts shouldn't we be using `admin_enqueue_scripts`?

3. Are these scripts being used in any front-end blocks or just the admin? If just admin - we should consider changing these class names to be more specific like `LoadAdminCampaignScripts` and if necessary additional `LoadPublicCampaignScripts` for anything front facing